### ISSUE TITLE
Albums with same title get merged.

### ISSUE THREAD

**Issue by Deses** [simulot/immich-go#19]
I'm not sure where to report this, but I think it might be more relevant here.

I used your tool to import a Google Takeout zip file using this command and it successfully ran the entire upload process without complaining.
`immich-go -server=URL -key=TOKEN -log-level=INFO upload -google-photos takeout-20230922T153302Z-001.zip`

Now, double checking the albums I detected a problem with different albums that have the **same name**.

In `Google Photos` I have 42 albums, and  `immich-go` created 40. 

For instance I have 2 `Viaje a León`, one from 2015 and another from 2017, and they both got merged.

I'm reporting this because you may have overlooked this, or maybe you really don't have a way to differentiate between albums with the same name. If that's the case, adding a pre-requisite that users should make sure there are no albums with the same name would be nice to have.

**Comment by simulot** [comment]
Hi!
It's an issue... so this is the good place to report it.

I haven't  anticipated your case. Each users have different use cases... At the moment, albums are identified by their name, hence the merging.

In the takeout archives albums are stored into directories having the name of the album. In the directory, there is a metadata file that contains the original name of the album. My guess is that album's folder are slightly different, but the album name in the json files are identical/

- Could you confirm the name of the metadata.json file? I know there are localized names for them. 
- Could explain how albums with same names are stored int the archive? their folder name?

Thanks.

**Comment by Deses** [comment]
Yes! Of course! I'd like to help you make immich-go as rock solid as possible. :)

In the Takeout archive, albums with the same name look like this:
![image](https://github.com/simulot/immich-go/assets/1654959/39807867-7a97-41c7-a44c-ae2c0751dbb9)
`Sin título = Untitled`

Each album doesn't have a `metadata.json` file, instead Google in their infinite wisdom produced a spanish-translated `metadatos.json`.

Here they are so you can take a look. Their structure inside looks normal and in english, thankfully.
[Sample metadata.zip](https://github.com/simulot/immich-go/files/12706016/Sample.metadata.zip)

Probably the thing that will be useful to you is:
```
"date": {
    "timestamp": "1496844125",
    "formatted": "7 jun 2017, 14:02:05 UTC"
  },
```

**Comment by simulot** [comment]
> Yes! Of course! I'd like to help you make immich-go as rock solid as possible. :)

Great! I love that attitude! Deal!

> Sin título = Untitled
Oops... I was thinking untitled albums was some remains of some photo sharing.  Should I keep them?


> Each album doesn't have a metadata.json file, instead Google in their infinite wisdom produced a spanish-translated metadatos.json.

I have anticipated this. But I prefer to check.  I'm wondering what could be the name for calatan or galican...

> Here they are so you can take a look. Their structure inside looks normal and in english, thankfully.
[Sample metadata.zip](https://github.com/simulot/immich-go/files/12706016/Sample.metadata.zip)

Thanks. Hopefully, JSON names remains identical across all languages I have encountered so far.
I have seen that album's folder names differs but not their titles in the json.

**Comment by Deses** [comment]
> Oops... I was thinking untitled albums was some remains of some photo sharing. Should I keep them?
I do have photos in 2 of them, and others are empty. IMHO I would keep them, the user is always free to keep or delete them using Immich's UI.

> I have anticipated this. But I prefer to check. I'm wondering what could be the name for calatan or galican...

I have more Google accounts lying around, I could change the language and test.

> Thanks. Hopefully, JSON names remains identical across all languages I have encountered so far.
I have seen that album's folder names differs but not their titles in the json.

Imagine if every language translated the JSON structure... that would be absolute madness. 💀

**Comment by Deses** [comment]
Indeed it is `metadades.json` in catalan.

**Comment by simulot** [comment]
> Indeed it is `metadades.json` in catalan.

🤨 why not... I'll add it to the list of possible names.
I have to add an option to give the name used in takeout...or guess  then name

**Comment by Deses** [comment]
I'd add a supported list of languages, and create a permanent issue where people can request more languages added, and maybe explain how to manually add more languages by the user.

**Comment by simulot** [comment]
Here are some fresh news:
I have refactored the google photo import. I expect this close most of opened issues.

I have found a way to detect the metadata.json file what ever its language. It's now based on the JSON structure.

I have managed the untitled album issue. Now, I'm using the folder name when the title found in the album JSON is not set.

I think there is no way to prevent the merge of album having the same name.
Example:
In the folder  "Viaje a León", the title in the metadada file is "Viaje a León"
In the folder "Viaje a León(1)", the title in the metadada file is also  "Viaje a León"

Let me know if this is a problem to have the both album merged into the immich album's "Viaje a León"

**Comment by Deses** [comment]
Can you compare if the metadata name and the folder are named the same?

I think this is a problem, because these names are autogenerated by Google Photos (I think? I don't think my father would take time to make albums) so there are potentially a lot of duplicated album names out there.

**Comment by simulot** [comment]
I see... Not everyone is giving a name stating by YYYY-MM-DD to his files/folder/albums  to have them sorted automatically... :-)

The metadata file (whatever its name) indicates that the folder is actually an album. But it's content is less relevant than the folder name itself. So I'll name the album after the name of the folder ignoring the content of the metadata.

**Comment by simulot** [comment]
The album takes always the folder name
closed with release v0.3.1

**Comment by Deses** [comment]
I'm going to test this right now!
